# Batterygate example (not actual values)

> 2020/05/07

We assume our internal battery to have a emf of $\varepsilon = 3.7\text{V}$ and that the electronics at $V_{ext}$ needs at leat $3.3 \text{V}$. The maximum power consumption of the iPhone is considered at $2 \text{Watt}$. Determine the maximum internal resistance $r$ of the battery.

- $\varepsilon = 3.7\text{V}~~~~~~~~~~~~$internal potential difference of the battery
- $V_{ext} \gt 3.3 \text{V}~~~~~~~~$minimum external voltage for electronics
- $P_{max} = 2 \text{W}~~~~~~~~$maximum power of smartphone

__Formula__
- $P = VI$
- $R = \frac{V}{I}$
- $\varepsilon = I(R + r)$

__Answer:__ The maximum internal resistance of the battery can be $0.66 \Omega$.

In [1]:
emf   = 3.7     # internal potential difference of the battery in Volt
V_ext = 3.3     # minimal external voltage at the electronics
P     = 2.0     # maximum power of smartphone electronics

I = P / V_ext   # maximum current at maximum power for minimum voltage
V_int = emf - V_ext # voltage drop over internal resistance
r = V_int / I   # maximum internal resistance

print("The maximum internal resistance of the battery can be {:.2f} Ohm.".format(r))

The maximum internal resistance of the battery can be 0.66 Ohm.


After two years the internal resistance has risen to $r = 2\Omega$. Determine $V_{ext}$. 

In [2]:
emf   = 3.7     # internal potential difference of the battery in Volt
V_ext = 3.3     # minimal external voltage at the electronics
P     = 2.0     # maximum power of smartphone electronics
r     = 2.0     # internal resistance of battery after two years, in Ohm  

R = V_ext**2 / P   # respective resistance of smartphone at max power for new battery
I = emf / (R + r)  # current in old battery with higher internal resistance
V_electronics = R * I   # voltage at the electronics

print("With an old battery and internal resistance of 2 Ohm the electronics gets only {:.2f} Volt.".format(V_electronics))

With an old battery and internal resistance of 2 Ohm the electronics gets only 2.71 Volt.


$2.71 \text{Volt}$ is not enough to power the smartphone. At high demand it shuts of.

A software feature now limits the maximum power consumption of the smartphone to $P = 0.6 \text{W}$ until the battery is changed. Determine $V_{ext}$ with this software update.

In [3]:
emf   = 3.7     # internal potential difference of the battery in Volt
V_ext = 3.3     # minimal external voltage at the electronics
P     = 0.6     # maximum power of smartphone electronics
r     = 2.0     # internal resistance of battery after two years, in Ohm  

R = V_ext**2 / P   # respective resistance of smartphone at new max power
I = emf / (R + r)  # current in old battery with power limit for electronics
V_electronics2 = R * I   # voltage at the electronics

print("With an old battery and a power limit of 0.6 W the electronics gets {:.2f} Volt.".format(V_electronics2))

With an old battery and a power limit of 0.6 W the electronics gets 3.33 Volt.
